zbarcam/zbarcam.c

```diff
@@ -132,6 +132,10 @@ static void data_handler (zbar_image_t *img, const void *userdata)
                 continue;
         }
         else if(format == RAW) {
+#ifdef _WIN32
+            _setmode(_fileno(stdout), _O_BINARY);
```

Comment:
You should check the return value.

Reply:
The return value of `_setmode`? To be honest, I just copied from later in the file (around line 315) where it's used like this:
```
    if(format == XML) {
#ifdef _WIN32
        fflush(stdout);
        _setmode(_fileno(stdout), _O_BINARY);
#endif
        printf(xml_head, video_device);
        fflush(stdout);
    }
```
Should that check the return value too?

Reply:
Yes, it should. It would be better to merge both calls in `main()`.

Reply:
Something more like this?

https://github.com/itsmattkc/zbar/blob/win32-bin-fix/zbarcam/zbarcam.c#L311

I notice `zbarimg.c` doesn't check `_setmode`'s return value either. If this looks good, I can write a similar check into zbarimg and PR both

Reply:
You should change:
```diff
-            fprintf(stderr, "ERROR: failed to set stdout mode: %i\n\n", errno);
-            return(usage(1));
+            fprintf(stderr, "ERROR: failed to set stdout mode: %i\n", errno);
+            return(1);
```

Reply:
Done

Reply:
Certainly.